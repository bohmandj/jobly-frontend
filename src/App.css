.App {
  --main-bg-color: #282c34;
  --nav-bg-color: #141c2b;
  text-align: left;
}