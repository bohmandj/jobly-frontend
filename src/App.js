import React from "react";
import Jobly from './Jobly';
import './App.css';
import './Jobly.css';

function App() {
  return (
    <div className="App">
      <Jobly />
    </div>
  );
}

export default App;
