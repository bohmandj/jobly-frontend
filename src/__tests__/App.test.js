import React from 'react';
import {
  render,
  screen,
  waitFor
} from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";
import App from '../App';

describe('App', () => {
  it("renders without crashing", async () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
  });

  it("matches snapshot", async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });

  it("loads homepage and displays landing page correctly", async () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    await waitFor(() => expect(screen.getByText(/All the jobs in one, convenient place/i)).toBeInTheDocument());
  });
})