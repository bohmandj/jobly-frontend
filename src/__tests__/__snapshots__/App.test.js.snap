// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Jobly"
    >
      <nav
        class="navbar navbar-expand-md fixed-top"
      >
        <div
          class="container-fluid"
        >
          <a
            aria-current="page"
            class="navbar-brand active"
            href="/"
          >
            Jobly
          </a>
          <ul
            class="ml-auto nav-grid navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class=""
                href="/login"
              >
                Log in
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class=""
                href="/signup"
              >
                Sign up
              </a>
            </li>
          </ul>
        </div>
      </nav>
      <div
        class="homepage my-auto"
      >
        <div
          class="container text-center"
        >
          <h1>
            Jobly!
          </h1>
          <p>
            All the jobs in one, convenient place.
          </p>
          <div>
            <button
              class="btn btn-primary fw-bold me-3 col-2"
            >
              Log in
            </button>
            <button
              class="btn btn-primary fw-bold col-2"
            >
              Sign up
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
